frankenphp.c

```diff
@@ -509,12 +511,63 @@ static char* frankenphp_read_cookies(void)
 	return ctx->cookie_data;
 }
 
-void frankenphp_register_bulk_variables(char **variables, size_t size, zval *track_vars_array)
+static void frankenphp_register_known_variable(const char *key, char *value, zval *track_vars_array, bool f)
+{
+	if (value == NULL) {
+		return;
+	}
+
+	php_register_variable(key, value, track_vars_array);
+	if (f) {
+		free(value);
+		value = NULL;
+	}
+}
+
+void frankenphp_register_bulk_variables(char *known_variables[27], char **dynamic_variables, size_t size, zval *track_vars_array)
 {
-	for (size_t i = 1; i < size; i = i+2)
+	/* Not used, but must be present */
+	frankenphp_register_known_variable("AUTH_TYPE", "", track_vars_array, false);
+	frankenphp_register_known_variable("REMOTE_IDENT", "", track_vars_array, false);
+
+	/* Allocated in frankenphp_update_server_context() */
+	frankenphp_register_known_variable("CONTENT_TYPE", (char *) SG(request_info).content_type, track_vars_array, false);
+	frankenphp_register_known_variable("PATH_TRANSLATED", (char *) SG(request_info).path_translated, track_vars_array, false);
+	frankenphp_register_known_variable("QUERY_STRING", SG(request_info).query_string, track_vars_array, false);
+	frankenphp_register_known_variable("REMOTE_USER", (char *) SG(request_info).auth_user, track_vars_array, false);
+	frankenphp_register_known_variable("REQUEST_METHOD", (char *) SG(request_info).request_method, track_vars_array, false);
+	frankenphp_register_known_variable("REQUEST_URI", SG(request_info).request_uri, track_vars_array,false);
+
+	/* Known variables */
+	frankenphp_register_known_variable("CONTENT_LENGTH", known_variables[0], track_vars_array, true);
```

Comment:
It might be worth using consts for the indices of `known_variables` and then reusing them in `allocServerVariable()` on the Go side. It'd help with maintaining and reviewing, but other than that, wouldn't do anything.

Reply:
I'm not sure it will help maintenance to add macros, to be honest, but PR is welcome!